lib/AST/Decl.cpp

```diff
@@ -7106,6 +7106,10 @@ void AbstractFunctionDecl::prepareDerivativeFunctionConfigurations() {
 ArrayRef<AutoDiffConfig>
 AbstractFunctionDecl::getDerivativeFunctionConfigurations() {
   prepareDerivativeFunctionConfigurations();
+  // Check `@differentiable` attributes.
+  for (auto *diffAttr : getAttrs().getAttributes<DifferentiableAttr>())
+    (void)diffAttr->getParameterIndices();
```

Comment:
What does this do?  Is it side-effecting?

Reply:
Yes, it triggers the request that performs `@differentiable` attribute type-checking.

When `@differentiable` attribute type-checking is further requestified, we'll probably use a different API to trigger the request instead of `diffAttr->getParameterIndices()`.

Reply:
Is there a better naming convention for getters that perform type checking requests?  The current name doesn't imply any side effects.

Reply:
Probably. I'm not too concerned about that right now, we'll be forced to find a better name when `@differentiable` attribute type-checking is further requestified.

Reply:
I'm a bit concerned with the readability.  I would write a comment at call sites like this where the result is being discarded. In this case, the comment "Check `@differentiable` attributes" is very unclear as it does not explain why you merely call a getter in order to check attributes.

Reply:
Thanks, I improved the comment.
I checked other similar usages of `getParameterIndices`, they're already documented: [[0]](https://github.com/apple/swift/blob/fae995ae26fdc241326dbc76d0987a288f5fe1c4/lib/SIL/IR/SILFunctionBuilder.cpp#L69-L77), [[1]](https://github.com/apple/swift/blob/fae995ae26fdc241326dbc76d0987a288f5fe1c4/lib/Sema/TypeCheckProtocol.cpp#L336-L346)